src/sidebar/config/test/build-settings-test.js

```diff
@@ -149,18 +149,20 @@ describe('sidebar/config/fetch-config', () => {
           appType: 'via',
         };
         fakeJsonRpc.call.onFirstCall().resolves({ foo: 'baz' }); // host config
-        const result = await fetchConfig(appConfig, fakeWindow);
+        const result = await buildSettings(appConfig, fakeWindow);
         assert.deepEqual(result.services[0].groups, ['group1', 'group2']);
       });
 
       it("creates a merged config where `groups` is a promise when its initial value is '$rpc:requestGroups'", async () => {
         const appConfig = {
-          services: [{ groups: '$rpc:requestGroups' }],
           appType: 'via',
         };
-        fakeJsonRpc.call.onFirstCall().resolves({ foo: 'baz' }); // host config
+        fakeJsonRpc.call.onFirstCall().resolves({
+          foo: 'baz',
+          services: [{ groups: '$rpc:requestGroups' }],
+        }); // host config
         fakeJsonRpc.call.onSecondCall().resolves(['group1', 'group2']); // requestGroups
```

Comment:
This test change and the change to the subsequent test is explained in this comment: https://github.com/hypothesis/client/pull/4242#discussion_r811381077